crates/poolnhl_infrastructure/src/services/players_service.rs

```diff
@@ -15,14 +16,35 @@ use crate::database_connection::DatabaseConnection;
 
 #[derive(Clone)]
 pub struct MongoPlayersService {
-    db: DatabaseConnection,
+    collection: Collection<PlayerInfo>,
 }
 
 impl MongoPlayersService {
     pub fn new(db: DatabaseConnection) -> Self {
-        Self { db }
+        let collection = db.collection::<PlayerInfo>("players");
+        Self { collection }
     }
 }
+
+pub async fn get_player_with_id(
+    collection: &Collection<PlayerInfo>,
+    player_id: i64,
+) -> Result<PlayerInfo> {
+    // Moved this part of the function here cause it is being reused in draft-service,
+    // not only in player-services.
+    let filter = doc! {"id": player_id};
+
+    let player = collection
+        .find_one(filter, None)
+        .await
+        .map_err(|e| AppError::MongoError { msg: e.to_string() })?
+        .ok_or_else(|| AppError::CustomError {
+            msg: format!("Player with id {} not found", player_id),
+        })?;
+
+    Ok(player)
```

Comment:
⛏️ Isn't this equivalent to
```suggestion
return  collection
        .find_one(filter, None)
        .await
        .map_err(|e| AppError::MongoError { msg: e.to_string() })?
        .ok_or_else(|| AppError::CustomError {
            msg: format!("Player with id {} not found", player_id),
        });
```